enterprise_gateway/services/processproxies/yarn.py

```diff
@@ -108,49 +111,67 @@ def launch_process(self, kernel_cmd, **kwargs):
 
         """
 
-    def confirm_yarn_queue_avaibility(self, **kwargs):
+    def confirm_yarn_queue_availability(self, **kwargs):
         """
         confirms if the yarn queue has capacity to handle the resource requests that
```

Comment:
Please capitalize 'confirms' -> 'Confirms'.

I think we should also add a statement near the beginning that this algorithm is subject to change as use cases become more defined.  That then would allow the removal of "Current version of check," from the second bullet below.